fmt/posix.cc

```diff
@@ -61,6 +61,27 @@ typedef ssize_t RWResult;
 
 inline std::size_t convert_rwcount(std::size_t count) { return count; }
 #endif
+
+std::FILE* u8fopen(fmt::CStringRef filename, fmt::CStringRef mode) {
+#ifdef _WIN32
+  fmt::UTF8ToUTF16 nconv(filename.c_str());
+  fmt::UTF8ToUTF16 mconv(mode.c_str());
+  return FMT_SYSTEM(_wfopen(nconv.c_str(), mconv.c_str()));
+#else
+  return FMT_SYSTEM(fopen(filename.c_str(), mode.c_str()));
+#endif
+}
+
+int u8open(fmt::CStringRef filename, int flags, int mode) {
+#ifdef _WIN32
+  fmt::UTF8ToUTF16 conv(filename.c_str());
+  int fd = -1;
+  FMT_POSIX(wsopen_s(&fd, conv.c_str(), flags, _SH_DENYNO, mode));
+  return fd;
+#else
+  return FMT_POSIX_CALL(open(filename, flags, mode));
```

Comment:
This doesn't compile because `open` expects a C string. BTW why not pass `const char *` instead of `CStringRef` here and in `u8fopen` to avoid calls to `c_str`?